src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -1358,24 +1358,25 @@ trait Definitions extends api.StandardDefinitions {
       clazz setInfo GenPolyType(List(tparam), ClassInfoType(parents, newScope, clazz)) markAllCompleted
     }
 
-    def newPolyMethod(typeParamCount: Int, owner: Symbol, name: TermName, flags: Long)(createFn: PolyMethodCreator): MethodSymbol = {
+    def newPolyMethod(typeParamCount: Int, owner: Symbol, name: TermName, flags: Long, enter: Boolean = true)(createFn: PolyMethodCreator): MethodSymbol = {
       val msym    = owner.newMethod(name.encode, NoPosition, flags)
       val tparams = msym.newSyntheticTypeParams(typeParamCount)
       val mtpe    = createFn(tparams) match {
         case (Some(formals), restpe) => MethodType(msym.newSyntheticValueParams(formals), restpe)
         case (_, restpe)             => NullaryMethodType(restpe)
       }
-
-      msym setInfoAndEnter genPolyType(tparams, mtpe) markAllCompleted
+      msym setInfo genPolyType(tparams, mtpe)
+      if (enter) msym.owner.info.decls enter msym
+      msym.markAllCompleted()
     }
 
     /** T1 means one type parameter.
      */
-    def newT1NullaryMethod(owner: Symbol, name: TermName, flags: Long)(createFn: Symbol => Type): MethodSymbol = {
-      newPolyMethod(1, owner, name, flags)(tparams => (None, createFn(tparams.head)))
+    def newT1NullaryMethod(owner: Symbol, name: TermName, flags: Long, enter: Boolean = true)(createFn: Symbol => Type): MethodSymbol = {
+      newPolyMethod(1, owner, name, flags, enter)(tparams => (None, createFn(tparams.head)))
     }
-    def newT1NoParamsMethod(owner: Symbol, name: TermName, flags: Long)(createFn: Symbol => Type): MethodSymbol = {
-      newPolyMethod(1, owner, name, flags)(tparams => (util.SomeOfNil, createFn(tparams.head)))
+    def newT1NoParamsMethod(owner: Symbol, name: TermName, flags: Long, enter: Boolean = true)(createFn: Symbol => Type): MethodSymbol = {
+      newPolyMethod(1, owner, name, flags, enter)(tparams => (util.SomeOfNil, createFn(tparams.head)))
     }
```

Comment:
when you add enterNewT1 etc, you can reduce vertical space by removing these extra braces! /protip :)